libs/networking/Http_mock_client.mli

```diff
@@ -66,6 +73,13 @@ val check_header : Cohttp_lwt.Request.t -> string -> string -> unit
   * Example: [check_header request "Authorization" "Bearer <token>"]
   *)
 
+val check_headers : Cohttp.Header.t -> Cohttp.Header.t -> unit
+(** [check_header request header header_value] will use Alcotest to assert
+  * two lists of headers are equal. Note that order and header capitlisation
```

Comment:
Nit: Should we ignore capitalization? Imo no, I assume most servers don't care about header capitalization, but also that'd really suck to debug if they do

Reply:
Didn't know it was part of the RFC, totally agree then

Reply:
Current ref for posterity: https://www.rfc-editor.org/rfc/rfc9110.html#section-5.1-3